.travis.yml

```diff
@@ -23,11 +21,7 @@ matrix:
     - php: nightly
   fast_finish: true
 
-install:
-  - composer self-update -q
-  - if [ -z "$deps" ]; then composer install -n; fi;
-  - if [ "$deps" = "low" ]; then composer update --prefer-lowest --prefer-stable --prefer-source -n; fi;
-  - if [ "$deps" = "high" ]; then composer update --prefer-source -n; fi;
+install: composer update $COMPOSER_FLAGS -n
```

Comment:
According to @stof and @GrahamCampbell, self-update is not required :) 

I think it's required only if we use features of lastest versions of Composer and TravisCI version is recent enough to support `--prefer-lowest` and `--prefer-stable`. It only adds useless build time.